input/pagecontent/hdecl-api.md

```diff
@@ -0,0 +1,1279 @@
+muutmise sõnumi. UPTIS kontrollib päringu tegija õiguseid, andmete suletust ja vajaduse korral edastab vastuseks 
+TD uue versiooni eeltäidetud andmetega: kui TD muutub staatusest "completed"--> "in-progress" või kui on muudetud  
+TD kasutusala, siis sel juhul uuendatakse TD eeltäidetud andmetega, muudel juhtudel toimub TD-l andmete uuendamine 
+vajaduspõhiselt (terviseandmeid uuendatakse TD'l UPTIS-s süsteemses parameetris määratud aja tagant).
+
+### Päring
+- **URL**: `PUT [base]/QuestionnaireResponse/[id]`
+
+TODO Alljärgnevas näites puudub vastuste osa, muus osas peaks näide olema valiidne
+```
+PUT [base]/QuestionnaireResponse/31
+{
+  "resourceType": "QuestionnaireResponse",
+  "meta": {
+    "profile": ["https://fhir.ee/hdecl/StructureDefinition/ee-health-declaration"]
+  },
+  "language" : "et-EE",
+  "extension" : [
+    {
+      "extension" : [
+        {
+          "url" : "category",
+          "valueCoding" : {
+            "system" : "https://fhir.ee/CodeSystem/tervisekontrolli-kasutusala",
+            "code" : "military-service"
+          }
+        }
+      ],
+      "url" : "https://fhir.ee/hdecl/StructureDefinition/ee-health-declaration-category"
+    },
+    {
+      "extension" : [
+        {
+          "url" : "category",
+          "valueCoding" : {
+            "system" : "https://fhir.ee/CodeSystem/tervisekontrolli-kasutusala",
+            "code" : "driver-group-I"
+          }
+        }
+      ],
+      "url" : "https://fhir.ee/hdecl/StructureDefinition/ee-health-declaration-category"
+    }
+  ],
+  "questionnaire": "https://fhir.ee/example-questionnaire|0.2",
+  "status": "in-progress",
+  "subject": {"reference": "Patient/48806140249"},
+  "authored": "2024-12-10T11:00:00+02:00",
+  "author": {"reference": "Patient/48806140249"}
+}
+```
+
+
+### Vastus
+- **HTTP kood**: 201 Created
+- **Sisu**: Päringu päise parameetri `Prefer: return=representation` korral tagastatakse uus tervisedeklaratsioon (QuestionnaireResponse), mis sisaldab eeltäidetud andmeid.
+
+```
+HTTP/1.1 201 Created
+Location: [base]/QuestionnaireResponse/31/_history/2
+```
+
+## Tervisedeklaratsiooni kinnitamine
+UPTIS võtab vastu patsiendi või patsiendi esindaja poolt edastatud täitmisel ("in-progress") tervisedeklaratsiooni (TD)
+kinnitamise operatsiooni $complete. UPTIS kontrollib päringu tegija õiguseid, andmete suletust ja kontrollib, kas TD-l olevad terviseandmed
+on vahepeal muutunud. Kui terviseandmed ei ole muutunud, siis TD kinnitatakse. Kui terviseandmed on muutunud, siis tagastatakse vastus
+koos muutunud terviseandmetega ja antakse veateade. Patsient või patsiendi esindaja peab TD uuesti üle vaatama ja kinnitama.
+
+### Päring
+- **URL**: `POST [base]/QuestionnaireResponse/[id]/$complete`
+
+```
+POST [base]/QuestionnaireResponse/31/$complete
+{
+    "resourceType": "Parameters"
+}
+```
+
+### Vastus
+- **HTTP kood**: 200 OK
+
+```
+HTTP/1.1 200 OK
+```
+
+## Tervisedeklaratsiooni tühistamine
```

Comment:
It would be nice to add links to the OperationDefinition for each operation. i.e OperationDefinition-ee-health-declaration-cancel.html (relative is ok)